include/mbedtls/ssl.h

```diff
@@ -798,7 +814,14 @@ struct mbedtls_ssl_session
     unsigned char master[48];   /*!< the master secret  */
 
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
-    mbedtls_x509_crt *peer_cert;        /*!< peer X.509 cert chain */
+    mbedtls_x509_crt *peer_cert;       /*!< peer X.509 cert chain */
+#if !defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)
+    /*! The digest of the peer's end-CRT. This must be kept to detect CRT
+     *  changes during renegotiation, mitigating the triple handshake attack. */
+    unsigned char *peer_cert_digest;
+    size_t peer_cert_digest_len;
+    mbedtls_md_type_t peer_cert_digest_type;
```

Comment:
We need to store this explicitly because the hash might change across session resumption. The digest length is redundant, but I felt it was worth the 4 Bytes of RAM compared to the code to extract it.

Reply:
Ok, thanks for the clarification, I did not notice the point about session resumption. I'm happy with the code as it stands then.